### PR TITLE
Testable ELK implementation

### PR DESCRIPTION
**Pull request by tgross**
Kibana is a Node.js application that uses Elasticsearch as its upstream. We configure it with Containerbuddy to have an `onStart` and `onChange` handler that rewrites its config file to find the ES master-only nodes. The ES compose files are pulled from the `triton-elasticsearch` blueprint and self-configure in the same way that stack does.

TODO before this is ready to use for internal testing of Docker log drivers on Triton:
- [x] rework [`triton-nginx`](https://github.com/tgross/triton-nginx/pull/1) to update the top-level config
- [x] create a log-generating test client (using Nginx to send logs to logstash)
- [x] make a logstash configuration for parsing gelf inputs
- [x] update the test rig to let you easily test any/all of the log drivers separately
- [x] add some usage documentation

cc @misterbisson if you'd like to take a look
ref https://github.com/joyent/blueprints/issues/9

### CODE REVIEW

**Comment by tgross** [comment]
It looks like the latest version of Kibana expects Elasticsearch 2.1+, so I'll need to update https://github.com/tgross/triton-elasticsearch/pull/1 with that.

**Comment by tgross** [comment]
@misterbisson this should be ready for review at this point.

File: kibana/onStart.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+MASTER=
+while true
+do
+    # get the list of ES master-only nodes from Consul
+    MASTER=$(curl -Ls --fail http://consul:8500/v1/catalog/service/elasticsearch-master | jq -r '.[0].ServiceAddress')
```
Comment (misterbisson) [inline kibana/onStart.sh:7]: `consul:8500` assumed.

File: logstash/onStart.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+MASTER=
+while true
+do
+    # get the list of ES master-only nodes from Consul
+    MASTER=$(curl -Ls --fail http://consul:8500/v1/catalog/service/elasticsearch-master | jq -r '.[0].ServiceAddress')
```
Comment (misterbisson) [inline logstash/onStart.sh:7]: `consul:8500` assumed.

**Comment by misterbisson** [comment]
I'd merge it and pass it on to @askfongjojo and @joshwilsdo for their use.

**Pull request merged**